### DISCUSSION TITLE
Complete undefined references when defining new labels.

### DISCUSSION THREAD

**Discussion by gi1242** [latex-lsp/texlab#1026] [Ideas]
When I'm writing a document, I often don't number equations. Later, when I need it, I type:

    by equation~\ref{e:fancy-equation}

and then go back and label the equation with `\label{e:fancy-equation}`.

As soon as I type `\ref{e:fancy-equation}`, texlab flags it as an undefined reference. When I'm defining the label, it would be useful if texlab suggested current undefined references as completions to `\label{`.

Is this easy to implement? (I presume it may be useful to others with a similar work flow).

Thanks in advance!

**Comment by pfoerster** [comment]
@gi1242 This feature will be part of the next release 👍

**Reply by gi1242** [reply]
Super thanks so much :heart:
